### PR TITLE
WT-10380 Fix expected warning output in test_cursor_random

### PR DESCRIPTION
**Pull request by Sean04**
test_cursor_random started failing again on the macOS host after WT-10197 changed the warning message for eviction taking more than 1 minute. Not all the BFGs are linked to the ticket, there are others that appear in the logs of the other macOS failures. The test was initially failing on macOS due to stress on the host from running many tests concurrently. The original fix in WT-7492 was to ignore the warning as the expected output however that message is now outdated. I've fixed the test to reflect the new warning message.

In the WT-7492 it was discussed whether we should check for the masOS platform when ignoring the output. I've included it here but I'm not sure whether to keep that check.

### CODE REVIEW

**Comment by etienneptl** [comment]
@Sean04,
I like that we check the platform - we are aware of this for MacOS but we don't expect the same for other platforms.

File: test/suite/test_cursor_random.py [hunk 1]
```diff
@@ -113,7 +113,8 @@ def cursor_random_multiple_insert_records(self, n):
         # Ignore the eviction generation drain warning as it is possible for eviction to
         # take longer to evict pages due to overflow items on the page.
         self.conn.close()
-        self.ignoreStdoutPatternIfExists('evict generation drain waited')
+        if (sys.platform.startswith('darwin')):
+            self.ignoreStdoutPatternIfExists('Eviction took more than 1 minute')
```
Comment (etienneptl) [inline test/suite/test_cursor_random.py:117]: Can we define a variable for this specific string?

**Review by etienneptl — APPROVED** [review]
One comment which does not need another review.

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Comment by kommiharibabu** [comment]
@Sean04 Check the reason behind the test failure to find out if that has any relation to this PR.

**Comment by Sean04** [comment]
The test failure is unrelated to this change. It's an occurrence of [WT-10488](https://jira.mongodb.org/browse/WT-10488)

**Comment by Sean04** [comment]
Scheduling the macOS host in the PR testing.

**Comment by Sean04** [comment]
evergreen merge

**Pull request merged**